get_fm_data.py

```diff
@@ -0,0 +1,200 @@
+        self.log("FMdata, daily kickoff: start checking new EPEX prices in FM.")
+        self.attempts_today = 0
+        self.get_epex_prices()
+
+
+    def get_epex_prices(self, *args, **kwargs):
+        """ Communicate with FM server and check the results.
+
+        Request prices from the server
+        Make prices available in HA by setting them in input_text.epex_prices
+        Notify user if there will be negative prices for next day
+        """
+
+        # A max_number of retries (with a growing time gap) is used as it seems
+        # unlikely the data will become available after this period.
+        if self.attempts_today > self.max_attempts:
+            self.notify(f"Failed to get EPEX prices for {str((self.get_now() + timedelta(days=+1)).date())} from FM. Attempted ({self.attempts_today}) times today, wait for retry till tomorrow.")
+            return
+        self.log(f"FMdata, get_epex_prices, attempt {self.attempts_today}.")
+
+        self.authenticate_with_fm()
+        now = self.get_now()
+        startEPEX = str((now +timedelta(days=-1)).date())
+        #startEPEX = "2021-12-13"
+
+        url = self.args["fm_data_api"] + self.args["fm_data_api_epex"]
+        url_params = {
+            "event_starts_after": startEPEX + "T00:00:00.000Z",
+        }
+        res = requests.get(
+            url,
+            params=url_params,
+            headers={"Authorization": self.fm_token},
+        )
+
+        # Authorisation error, retry authoristion.
+        if res.status_code == 401:
+            self.handle_response_errors(message, res, "get EPEX prices", self.get_epex_prices, *args, **kwargs)
+            return
+        # Do not count authoristion errors int he attempts.
+        self.attempts_today += 1
+
+        # A bit prematurely calculate a delay in seconds in which to call
+        # this  function again incase something goes wrong.
+        # The delays total to 11h30m with max 12 attempts.
+        delay = (5 + self.attempts_today * 8) * 60
+
+        if res.status_code != 200:
+            self.log(f"Get FM EPEX data failed with status_code: {res.status_code}, full response was {res.json()}. Retry in {delay} minutes.")
+            self.run_in(self.get_epex_prices, delay)
+            return
+
+        prices = res.json()
+
+        # We might get older data, but if the UI has no data yet it is still
+        # nice to show this older data anayway, so process it.
+
+        # Not so usefull checks??
+        # epex_entity = self.get_state("input_text.epex_prices", attribute='records')
+        # if epex_entity != None:
+        #     #There are records already no need to process data
+        #     return
+        # self.log("Eventhough prices are not new, there are currently no prices registered at all (due to restart?), so process data anyway.")
+
+        # From FM format (€/MWh) to user desired format (€ct/kWh) 
+        # = * 100/1000 = 1/10. Also include VAT
+        VAT = float(self.args["VAT_NL"])
```

Comment:
This doesn't need to mention NL explicitly.

Reply:
I think you've missed my comments about adding `_NL` to arguments that don't need to be NL specific, but I'll open a new issue for that to not hold up this PR any longer.